gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/RequestCreatorTest.kt

```diff
@@ -119,6 +121,30 @@ class RequestCreatorTest {
         verify(picasso).cancelRequest(target)
     }
 
+    @Test
+    fun `getBitmap() should cancel Picasso request on main thread`() {
+        val thread = HandlerThread("").apply { start() }
+        assertTrue(thread.isAlive)
+        Dispatchers.setMain(Handler(thread.looper).asCoroutineDispatcher())
+        var cancelRequestThread: Thread? = null
+        val picasso = mock<Picasso> {
+            on { cancelRequest(any<Target>()) } doAnswer { cancelRequestThread = Thread.currentThread() }
```

Comment:
we capture the thread that `cancelRequest` is called on, that way we can assert later that it was the "Main" thread.

We can't directly throw an assertion here because `GlobalScope.launch` will swallow the exception and process it with an `UncaughtExceptionHandler` not attached to this test. The exception will print to the console, but the test will report as passing

Reply:
This might be worth putting in the comments of the test.